plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -874,18 +923,15 @@ public boolean downloadToFile(String urlString, Plugin plugin, File fileLocation
                     Files.delete(pluginFile.toPath());
                 }
             } catch (IOException e) {
-                logVerbose(String.format("Unable to delete %s before retry %d",pluginFile,i+1));
+                logVerbose(String.format("Unable to delete %s before retry %d", pluginFile, i + 1));
             }
 
             try (CloseableHttpClient httpclient = HttpClients.custom().useSystemProperties()
-                    .addInterceptorLast((HttpRequestInterceptor) (request, context) -> {
-                        throw new IOException("Retry on any failure");
```

Comment:
this was always causing an exception to be thrown before, which meant every request was retried and then it failed after 3 times